cli/cli/command_framework/highlevel/context_id_arg/context_id_arg.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/kurtosis-tech/kurtosis/contexts-config-store/api/golang/generated"
+	"github.com/kurtosis-tech/kurtosis/contexts-config-store/store"
+	"github.com/kurtosis-tech/stacktrace"
+	"sort"
+	"strings"
+)
+
+const (
+	uuidOrNameDelimiter = ", "
+
+	defaultIsRequired = false
+	defaultValueEmpty = ""
+)
+
+// NewContextIdentifierArg pre-builds context identifier arg which has tab-completion and validation ready out-of-the-box
+func NewContextIdentifierArg(
+	// The arg key where this context identifier argument will be stored
+	argKey string,
+	isGreedy bool,
+) *args.ArgConfig {
+
+	validate := getValidationFunc(argKey, isGreedy)
+
+	return &args.ArgConfig{
+		Key:                   argKey,
+		IsOptional:            defaultIsRequired,
+		DefaultValue:          defaultValueEmpty,
+		IsGreedy:              isGreedy,
+		ValidationFunc:        validate,
+		ArgCompletionProvider: args.NewManualCompletionsProvider(getCompletions),
+	}
+}
+
+// Make best-effort attempt to get context names
+func getCompletions(ctx context.Context, flags *flags.ParsedFlags, previousArgs *args.ParsedArgs) ([]string, error) {
+	contextsConfigStore := store.GetContextsConfigStore()
+
+	kurtosisContextsConfig, err := contextsConfigStore.GetKurtosisContextsConfig()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting Kurtosis contexts configuration")
+	}
+
+	var storedContextNames []string
+	for _, storedContextInfo := range kurtosisContextsConfig.GetContexts() {
+		storedContextNames = append(storedContextNames, storedContextInfo.GetName())
+	}
+
+	// we sort them individually
+	sort.Strings(storedContextNames)
+	return storedContextNames, nil
+}
+
+// Context identifier validation function
+func getValidationFunc(argKey string, isGreedy bool) func(context.Context, *flags.ParsedFlags, *args.ParsedArgs) error {
+	return func(ctx context.Context, flags *flags.ParsedFlags, args *args.ParsedArgs) error {
+
+		var contextIdentifiersToValidate []string
+		if isGreedy {
+			contextIdentifiers, err := args.GetGreedyArg(argKey)
+			if err != nil {
+				return stacktrace.Propagate(err, "Expected a value for greedy arg '%v' but didn't find any", argKey)
+			}
+			contextIdentifiersToValidate = append(contextIdentifiersToValidate, contextIdentifiers...)
+		} else {
+			contextIdentifier, err := args.GetNonGreedyArg(argKey)
+			if err != nil {
+				return stacktrace.Propagate(err, "Expected a value for non-greedy arg '%v' but didn't find one", argKey)
+			}
+			contextIdentifiersToValidate = append(contextIdentifiersToValidate, contextIdentifier)
+		}
+
+		_, err := GetContextUuidForContextIdentifier(contextIdentifiersToValidate)
+		if err != nil {
+			return stacktrace.Propagate(err, "Error finding context matching the provided identifiers")
+		}
+		return nil
+	}
+}
+
+func GetContextUuidForContextIdentifier(contextIdentifiers []string) (map[string]*generated.ContextUuid, error) {
```

Comment:
This could possibly benefit from a unit test with a mocked `ContextsConfigStore`

Reply:
That requires to slightly update how we pass in the contexts config store, but the more test the better I agree. I'll try to add